### ISSUE TITLE
Use WikiFont icons for Refreshed

### ISSUE THREAD

**Issue by MtMNC** [Brickimedia/brickimedia#344]
I'm not putting this in #274 since Refreshed 3.0 is already out. Do you think we should make a large issue for Refreshed suggestions, keep using #274, or just make smaller individual issues?

We should incorporate the [WikiFont](https://github.com/munmay/WikiFont) icons into Refreshed instead of the Font Awesome ones it uses currently. The actual font isn't going to be used in the MediaWiki core but the icons are being used in [Winter](https://www.mediawiki.org/wiki/Winter), [Flow](https://www.mediawiki.org/wiki/Flow), etc.

**Comment by georgebarnick** [comment]
Now that Refreshed 3.0 is fairly complete, and anything else remaining in #274 that hasn't been done will be a part of a subversion of refreshed 3.*, we can make any more suggestions as separate issues.

I definitely believe we should use WikiFont instead of Font Awesome. I'll take care of it.

**Comment by georgebarnick** [comment]
Do we want to use the font itself, or just use images of the characters?

**Comment by MtMNC** [comment]
I was thinking we'd use the vectors.

**Comment by MtMNC** [comment]
(Sorry I'm trying to figure out how assigning works.)

**Comment by georgebarnick** [comment]
Some problems:
- ```star``` and ```unStar``` are different colors than all the others as well as different size and position
- There is no icon for moving pages it seems. Using ```mapPin``` for now.
- I haven't changed the search, search close, sidebar expand, arrow down, or notification icons as they'll need further adjustment such as CSS sizing and positioning, and SVG recoloring.

**Comment by MtMNC** [comment]
Actually I found some black versions on the [MediaWiki page](https://www.mediawiki.org/wiki/Design/WikiFont)--I can implement those. As for the move icon, the [Font Awesome location-arrow](https://fortawesome.github.io/Font-Awesome/icon/location-arrow/) looks pretty similar to the [Winter move icon](http://unicorn.wmflabs.org/winter/images/icon_button_move_l_grey.png), so I think that could work at least temporarily.

**Comment by MtMNC** [comment]
Right, here's an attempt at a master list with all the WikiFont icons we could use to replace our current ones.
- [x] edit, etc. icons
- [ ] search icons ([open search](https://github.com/munmay/WikiFont/blob/master/svgs/uniE000%20-%20magnifyingGlass.svg), [close search](https://www.mediawiki.org/wiki/File:Close_icon.svg))
- [ ] [arrows](https://github.com/munmay/WikiFont/blob/master/svgs/uniE02865%20-%20caretDown.svg)
- [ ] user icons ([anon](https://www.mediawiki.org/wiki/File:SleepSpeechbubble_icon.svg), [logged in](https://www.mediawiki.org/wiki/File:AwakeSpeechbubble_icon.svg))
- [ ] [menu icon](https://www.mediawiki.org/wiki/File:Stripe_icon.svg)
- [ ] [notification icon?](https://www.mediawiki.org/wiki/File:Bell_icon.svg)

Also, seeing as GBC now has to use separate black and white icons for headers, etc., maybe using the actual WikiFont is a better idea after all. Vector is not designed to be customized, so all an icon font does is use extra space than individual images would--that might explain why Wikimedia isn't planning on including WikiFont with Vector. On the other hand, you're supposed to customize Refreshed, and using an icon font would significantly reduce the work required to do that.

Any thoughts/suggestions?

**Comment by georgebarnick** [comment]
With the point about GBC using other icon colors, and some other designs I've mocked up on Dev Wiki using other colors too, I think using the font actually would be smart.

And I believe any collaborator is able to check/uncheck the boxes.

**Comment by MtMNC** [comment]
Status update: I've added in all the icons except for the notification one and the arrow for the language dropdown (I don't have an interwiki setup on my local clone so I can't test it). However, I've broken a good deal of JS in the process, so I'm going in and reorganizing it (e.g. I'm implementing a .fadable class for all dropdowns so I can simply the JS for fading in/out with CSS transitions).

**Comment by MtMNC** [comment]
I'm done cleaning everything up, so everything should be good to go. I forgot to mention it before, but I also left out the menu icon since none of the WikiFont ones look that good (two are for a table of contents and the other is vertically squished). It'll of course require CSS changes to the wikis (especially Great Ball Contraption), so be forewarned before pushing it to the site.

https://github.com/Brickimedia/Refreshed/commit/1687371ade0a05b825c755c40d4517247e972e89

**Comment by MtMNC** [comment]
I've found and fixed several issues that exist in the current commit, but before I push it would anyone mind if I included jQuery Mobile's touch events? It has a tap event that circumvents issues with the "clickOrTouchend" event refreshed.js currently uses, plus it has swipe events that might come in handy. The jQuery Mobile site lets you download individual modules so we would only be including the piece that we need.
